foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiMessage.java

```diff
@@ -9,38 +10,78 @@
  * @since 1.4.0
  */
 @Beta
-public interface OpenAiMessage {
+public sealed interface OpenAiMessage
+    permits OpenAiUserMessage, OpenAiAssistantMessage, OpenAiSystemMessage, OpenAiToolMessage {
 
   /**
    * A convenience method to create a user message.
    *
-   * @param msg the message content.
+   * @param message the message content.
    * @return the user message.
    */
   @Nonnull
-  static OpenAiMessage user(@Nonnull final String msg) {
-    return new OpenAiUserMessage(msg);
+  static OpenAiUserMessage user(@Nonnull final String message) {
+    return new OpenAiUserMessage(message);
+  }
+
+  /**
+   * A convenience method to create a user message containing only an image.
+   *
+   * @param openAiImageItem the message content.
+   * @return the user message.
+   */
+  @Nonnull
+  static OpenAiUserMessage user(@Nonnull final OpenAiImageItem openAiImageItem) {
+    return new OpenAiUserMessage(new OpenAiMessageContent(List.of(openAiImageItem)));
   }
 
   /**
    * A convenience method to create an assistant message.
    *
-   * @param msg the message content.
+   * @param message the message content.
    * @return the assistant message.
    */
   @Nonnull
-  static OpenAiMessage assistant(@Nonnull final String msg) {
-    return new OpenAiAssistantMessage(msg);
+  static OpenAiAssistantMessage assistant(@Nonnull final String message) {
+    return new OpenAiAssistantMessage(message);
   }
 
   /**
    * A convenience method to create a system message.
    *
-   * @param msg the message content.
+   * @param message the message content.
    * @return the system message.
    */
   @Nonnull
-  static OpenAiMessage system(@Nonnull final String msg) {
-    return new OpenAiSystemMessage(msg);
+  static OpenAiSystemMessage system(@Nonnull final String message) {
+    return new OpenAiSystemMessage(message);
   }
+
+  /**
+   * A convenience method to create a tool message.
+   *
+   * @param message the message content.
+   * @return the tool message.
+   */
+  @Nonnull
+  static OpenAiToolMessage tool(@Nonnull final String message) {
+    return new OpenAiToolMessage(message);
+  }
```

Comment:
```suggestion
```
Tool messages need both id and content, also you said that this is out of scope so let's remove it.